app/controllers/tests_controller.rb

```diff
@@ -2,30 +2,39 @@ class TestsController < ApplicationController
   skip_before_action :check_admin, except: [:index]
 
   def create
-    course = Problem.find(params[:problem_id]).course
+    problem = Problem.includes(assignment: :course).find(params[:problem_id])
```

Comment:
The is a nested eager query. This is eagerly loads the problem's assignment, and the course of that assignment.